### ISSUE TITLE
Is there a way to specify a destination directory? And, if it isn't there, to create it?

### ISSUE THREAD

**Issue by plocket** [cpina/github-action-push-to-another-repository#8]
If we want the file to go in a specific directory in the destination repo, is there a way to specify that? If so, will it create the directory if the directory is missing?

**Comment by cpina** [comment]
Hi @plocket  - currently there isn't a way to specify the destination directory in the output repository but it's quite easy to add this feature some time soon. Would it help you?

**Comment by plocket** [comment]
Thank you for the help in the previous issue. I think this could be a useful feature though I'm all set for now with the combo I created. Thanks again for the help and thank you for the core code for what I needed.

**Comment by KomaGR** [comment]
I would like to +1 this feature request.

@plocket: What combo did you create to bypass this need?

**Comment by cpina** [comment]
@KomaGR : the github action, as it is now, it's deleting all the contents in the destination repository and then copying the generated files there. Would you still like that it deletes everything and creates a directory? Or in this case it should delete only the contents of the directory?

I'm thinking of adding this feature or not and if it would help in your use case.

**Comment by Laurianti** [comment]
I'm interested to have the possibility to publish on specific directory and/or to avoid the deletion of previous files.
Thank you.

**Comment by plocket** [comment]
@KomaGR : This is the code I've got right now and it's working ok for me: https://github.com/plocket/push-generated-file.

I forked this repo and used [@dmnemec's repo to make some changes](https://github.com/dmnemec/copy_file_to_another_repo_action). I also saw some more recent changes in here and built off of those too (thank you @cpina)

I have a destination folder input for the action here: https://github.com/plocket/push-generated-file/blob/master/action.yml#L13. Some of the bash file handling starts around here: https://github.com/plocket/push-generated-file/blob/master/entrypoint.sh#L49.

General steps:
1. Clone target repo
1. If the directory doesn't already exist, make it in the cloned repo: `mkdir -p`
1. Copy the files into the directory (for files beginning with `.` and for regular files)
1. `add` and `commit` the files if there are changes
1. Navigate into the cloned directory
1. Push the cloned directory to the target branch (creating the target branch if it doesn't exist)

Because it doesn't delete the existing repo or the existing files, the new files are just changes. It's not the cleanest code ever, but I don't code in bash much. Again, thank you for this work, @cpina.
